### PR TITLE
Interactive CLI interactive hook

### PR DESCRIPTION
**Pull request by dschep**
## What did you implement:

Added `interactiveCli:end` lifecycle hook to enable serverless/enterprise-plugin#226

## How did you implement it:
added `'end'` lifecycle hook to the interactiveCli plugin

## How can we verify it:
checkout this branch and use the plugin from the other pr and try the sls interactive flow, you should get the messages at the end.

## Todos:

_**Note: Run `npm run test-ci` to run all validation checks on proposed changes**_

- [ ] Write tests and confirm existing functionality is not broken.  
       **Validate via `npm test`**
- [ ] Write documentation
- [x] Ensure there are no lint errors.  
       **Validate via `npm run lint-updated`**  
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
- [x] Ensure introduced changes match Prettier formatting.  
       **Validate via `npm run prettier-check-updated`**  
       _Note: All reported issues can be automatically fixed by running `npm run prettify-updated`_
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

**_Is this ready for review?:_** YES  
**_Is it a breaking change?:_** NO

### CODE REVIEW

File: docs/providers/aws/guide/serverless.yml.md [hunk 1]
```diff
@@ -41,7 +41,12 @@ provider:
   logRetentionInDays: 14 # Set the default RetentionInDays for a CloudWatch LogGroup
   deploymentBucket:
     name: com.serverless.${self:provider.region}.deploys # Deployment bucket name. Default is generated by the framework
-    serverSideEncryption: AES256 # when using server-side encryption
+    serverSideEncryption: true # to use server-side encryption
+    sseKMSKeyId: arn:aws:kms:us-east-1:xxxxxxxxxxxx:key/aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa # when using server-side encryption
+    sseCustomerAlgorithim: AES256 # when using server-side encryption and custom keys
+    sseCustomerKey: string # when using server-side encryption and custom keys
+    sseCustomerKeyMD5: md5sum # when using server-side encryption and custom keys
+
```
Comment (medikoo) [inline docs/providers/aws/guide/serverless.yml.md]: This change looks unrelated

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "serverless",
-  "version": "1.50.0",
```
Comment (pmuens) [inline package.json]: We could keep the old version here to avoid cherry-picking as @medikoo mentioned in his PR review.
